services/kafkaService.go

```diff
@@ -0,0 +1,106 @@
+package services
+
+import (
+	"fmt"
+	"github.com/Shopify/sarama"
+	"github.com/companieshouse/chs-delta-api/config"
+	"github.com/companieshouse/chs-delta-api/models"
+	"github.com/companieshouse/chs.go/avro"
+	"github.com/companieshouse/chs.go/avro/schema"
+	"github.com/companieshouse/chs.go/kafka/producer"
+	"github.com/companieshouse/chs.go/log"
+)
+
+const (
+	SchemaName = "chs-delta"
+)
+
+// Used for unit testing. By Adding variables which link to certain package level functions / methods, we can
+// override them during unit testing and change them to point to mock implementations to assert functionality.
+var (
+	callSchemaGet   = schema.Get
+	callProducerNew = producer.New
+	callSend        = sendViaProducer
+)
+
+// KafkaService defines all Methods needed to successfully send a message onto a Kafka topic.
+type KafkaService interface {
+	Init(cfg *config.Config) error
+	SendMessage(topic, data string) error
+}
+
+// KafkaServiceImpl is a concrete implementation of the KafkaService interface.
+type KafkaServiceImpl struct {
+	schema string
+	P      *producer.Producer
+}
+
+// NewKafkaService returns a KafkaServiceImpl that isn't configured.
+func NewKafkaService() KafkaServiceImpl {
+	return KafkaServiceImpl{}
+}
+
+// Init initialises a KafkaService using a provided config.
+func (kSvc *KafkaServiceImpl) Init(cfg *config.Config) error {
+
+	// Retrieve the generic chs-delta avro schema.
+	log.Trace("Get schema from Avro", log.Data{"schema_name": SchemaName})
+	sch, err := callSchemaGet(cfg.SchemaRegistryURL, SchemaName)
+	if err != nil {
+		log.Error(fmt.Errorf("error receiving %s schema: %s", SchemaName, err))
+		return err
+	}
+	log.Trace("Successfully received schema", log.Data{"schema_name": SchemaName})
```

Comment:
Should we use InfoC here ? If we turn off Trace logging we won't be able to troubleshoot anything ?